### PR TITLE
merging multiple examples with similar objectives

### PR DESCRIPTION
**Pull request by tarun-mitruka**

### CODE REVIEW

File: src/iks001_computePi.cpp [hunk 1]
```diff
@@ -8,12 +8,62 @@
...
 #include <dune/geometry/quadraturerules.hh>
 #include <dune/grid/common/boundarysegment.hh>
+#include <dune/grid/io/file/vtk/vtkwriter.hh>
 
 #include <Eigen/Core>
 #include <Eigen/Dense>
 
 #include <ikarus/utils/drawing/griddrawer.hh>
 
+void globallyRefinedUnitCircle() {
```
Comment (rath3t) [inline src/iks001_computePi.cpp]: Can you rename this from global.. to something like naive or circle unaware refinement. I think local and global are unapproriate identifiers here.

File: src/iks001_computePi.cpp [hunk 2]
```diff
@@ -25,9 +75,9 @@ struct UnitCircleBoundary : Dune::BoundarySegment<2, 2, double> {
   std::array<Dune::FieldVector<double, 2>, 2> corners;
 };
 
-int main(int argc, char **argv) {
-  Dune::MPIHelper::instance(argc, argv);
-
+void locallyRefinedUnitCircle() {
```
Comment (rath3t) [inline src/iks001_computePi.cpp]: see above maybe here "boundaryAwareUnitCircle"

File: src/iks001_computePi.cpp (hunk 1 above)
Comment (tarun-mitruka) [reply src/iks001_computePi.cpp]: renamed to boundaryUnawareRefinedCircle()

File: src/iks001_computePi.cpp (hunk 2 above)
Comment (tarun-mitruka) [reply src/iks001_computePi.cpp]: renamed to boundaryAwareRefinedCircle()

**Review by rath3t — APPROVED** [review]
If this is done from your side, we can merge

**Pull request merged**